cohortextractor/query_engines/mssql.py

```diff
@@ -502,9 +552,15 @@ def get_sql(self):
         return "\n\n\n".join(sql)
 
     def query_expression_to_sql(self, query):
+        dialect = self.sqlalchemy_dialect.dialect()
+        # SQLAlchemy will set this for us when connecting to an actual database
+        # (it asks for the server version and then updates capabilities
+        # accordingly) but we haven't connected to a database here so we have
+        # to set it manually
+        dialect.supports_multivalues_insert = True
```

Comment:
I think there's definitely an argument for asserting that that property is supported at the point we actually connect. It would feel a bit annoying to me though (for debugging purposes) to structure things so that you can't even see the generated queries without having a real database to connect to.

Reply:
Yes, that would be annoying. Fair enough.